OpenRA.Mods.Common/Activities/DeliverResources.cs

```diff
@@ -43,7 +43,7 @@ protected override void OnFirstRun(Actor self)
 
 		public override bool Tick(Actor self)
 		{
-			if (IsCanceling)
+			if (IsCanceling || harv.IsTraitDisabled)
```

Comment:
I believe we should also be calling `INotifyHarvesterAction.MovementCancelled` here, so pull this into its own case that calls `Cancel`?